WordPress/src/main/java/org/wordpress/android/ui/stats/StatsSinglePostDetailsActivity.java

```diff
@@ -0,0 +1,711 @@
+            mSwipeToRefreshHelper.setRefreshing(false);
+            ToastUtils.showToast(this, this.getString(R.string.connection_error), ToastUtils.Duration.LONG);
+            return;
+        }
+
+        final RestClientUtils restClientUtils = WordPress.getRestClientUtilsV1_1();
+        final String blogId = mRemotePostItem.getBlogID();
+
+        // View and visitor counts for a site
+        final String singlePostRestPath = String.format(
+                "/sites/%s/stats/post/%s", blogId, remotePostID);
+
+        AppLog.d(AppLog.T.STATS, "Enqueuing the following Stats request " + singlePostRestPath);
+
+        RestBatchCallListener vListener = new RestBatchCallListener(this);
+        restClientUtils.get(singlePostRestPath, vListener, vListener);
+
+        mIsUpdatingStats = true;
+        mSwipeToRefreshHelper.setRefreshing(true);
+
+        mMonthsAndYearsList.setVisibility(View.GONE);
+        mAveragesList.setVisibility(View.GONE);
+        mRecentWeeksList.setVisibility(View.GONE);
+
+    }
+
+    private void setupEmptyUI(boolean isLoading) {
+        Context context = mGraphContainer.getContext();
+        if (context != null) {
+            LayoutInflater inflater = LayoutInflater.from(context);
+            View emptyBarGraphView = inflater.inflate(R.layout.stats_bar_graph_empty, mGraphContainer, false);
+            final TextView emptyLabel = (TextView) emptyBarGraphView.findViewById(R.id.stats_bar_graph_empty_label);
+            if (isLoading) {
+                emptyLabel.setText("");
+            } else {
+                emptyLabel.setText(getString(R.string.stats_bar_graph_empty));
+            }
+            if (emptyBarGraphView != null) {
+                mGraphContainer.removeAllViews();
+                mGraphContainer.addView(emptyBarGraphView);
+            }
+        }
+        mStatsViewsLabel.setText("");
+        mStatsViewsTotals.setText("");
+        mMonthsAndYearsList.setVisibility(View.GONE);
+        mAveragesList.setVisibility(View.GONE);
+        mRecentWeeksList.setVisibility(View.GONE);
+        mAveragesIdToExpandedMap.clear();
+        mYearsIdToExpandedMap.clear();
+    }
+
+    private VisitModel[] getDataToShowOnGraph () {
+        if (mRestResponseParsed == null) {
+            return new VisitModel[0];
+        }
+
+        final VisitModel[] dayViews = mRestResponseParsed.getDayViews();
+        if (dayViews == null) {
+            return new VisitModel[0];
+        }
+
+        int numPoints = Math.min(getNumOfPoints(), dayViews.length);
+        int currentPointIndex = numPoints - 1;
+        VisitModel[] visitModels = new VisitModel[numPoints];
+
+        for (int i = dayViews.length - 1; i >= 0 && currentPointIndex >= 0; i--) {
+            visitModels[currentPointIndex] = dayViews[i];
+            currentPointIndex--;
+        }
+
+        return visitModels;
+    }
+
+    private int getNumOfPoints() {
+        if(StatsUIHelper.shouldLoadMoreBars(this)) {
```

Comment:
These numbers are the same as those used on [StatsVisitorsAndViewsFragment](https://github.com/wordpress-mobile/WordPress-Android/blob/issue/2153-stats-single-post-details-page/WordPress/src/main/java/org/wordpress/android/ui/stats/StatsVisitorsAndViewsFragment.java#L430) so should be constants rather than hard-coded.